src/main/java/codeu/model/data/MentionList.java

```diff
@@ -0,0 +1,99 @@
+package codeu.model.data;
+
+import codeu.model.store.basic.UserStore;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class MentionList {
+    private Message message;
+    private  UserStore userStore;
+
+    public MentionList(Message message, UserStore userStore){
+        message = this.message;
+        userStore = this.userStore;
+    }
+
```

Comment:
...and add a public static factory method:
public MentionParser createParser(Message message, UserStore userStore) {
  return new MentionParser(message, userStore);
}

This will make it possible to later alter the parsing logic or return a specialized subclass of this without changing the type -- it's almost like using an interface as a type, and that can be especially handy when working with tests and with purely logic classes like this (logic class: one that mostly just implements logic, contrasted with a data class: one that mostly stores data).